### DISCUSSION TITLE
Only listen to certain senders in hub to hub communication?

### DISCUSSION THREAD

**Discussion by Potato-eater** [pybricks/support#1250] [Q&A]
I wanted my robot to listen to only certain hubs, and would ignore other hubs. Is there any way to achieve this?

**Comment by laurensvalk** [comment]
Yes, set the `observe_channels` to match only the broadcast channels used by the hubs you want to listen to.

**Reply by JDwyer009** [reply]
@Potato-eater 
Last weekend, I was at the RCJA's National Competition in Aust promoting the Bluetooth functionality available in Pybricks. There was a lot of interest for port expansion and robot communication purposes. The Bluetooth communication works flawlessly which is sensational. There are 255 channels to use. 

However, there are a couple of issues to consider:

1. It is a broadcast protocol, which means 1 to many communication is possible (this is awesome).
2. However, _**if**_ Pybricks were to release a Scratch-like version of their program ;) then it would likely become the dominant programming language used in competitions within a year or two due to the richness of its feature set.
3. While the plethora of 255 channels isn't an issue for home use or classroom use, it could be an issue at competitions with over 200 teams attending. The likelihood of two separate teams randomly broadcasting and listening on say channel 8 increases as the number go up. That, plus the ability for other teams to intentionally interfere on your teams channel.
4. It would seem one solution would be to add a unique variable/key at the beginning of your broadcast such that the listening hub would only execute the rest of the received code if the keys match.

my 2cents worth anyway.

**Reply by Potato-eater** [reply]
I was actually competing in that exact competition! I was the team who won soccer standard.

The reason why I asked this question is that there seemed to be some interference in the Bluetooth communication during the competition.
We had a robot that had 2 hubs (so that we could have more sensors), and used Bluetooth to communicate between them. During the matches though, someone seemed to be sending random signals in all the channels. We tried to change the channel we use for 2 times, but both of them are also affected. So we had to use only one hub during the semifinals (but we still won!). 
My robot were coded so that it would work even only with one hub. Like you said, I had a password and encodings on the values I send, so those interference wasn't a big problem for me, but 2 other teams were less fortunate. their robots would crash whenever they received those random signals.

Even when I was using the passwords, the interference signals would still slightly slow down the robot since you can only listen to one message at a time.

**Reply by laurensvalk** [reply]
> However, _**if**_ Pybricks were to release a Scratch-like version of their program ;) then it would likely become the dominant programming language used in competitions within a year or two due to the richness of its feature set.

:eyes:
